### ISSUE TITLE
[Distribution-build-opensearch Breaks] Error building alerting components

### ISSUE THREAD

**Issue by zelinh** [opensearch-project/opensearch-build#1662]
Currently our distribution-build-opensearch is broken.
Here is the context of error.

```
A problem occurred evaluating root project 'opensearch-alerting'.
> A problem occurred configuring project ':alerting'.
  ​> Cannot mutate content repository descriptor 'MavenLocal(file:/usr/share/opensearch/.m2/repository/)' after repository has been used

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help [at https://help.gradle.](https://help.gradle.org/)org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
S[ee https://docs.gradle.org/6.6.1/userguide/command_line_interface.html#sec:command_line_warni](https://docs.gradle.org/6.6.1/userguide/command_line_interface.html#sec:command_line_warnings)ngs

BUILD FAILED in 16s
2022-02-21 21:11:15 ERROR    Error building alerting, retry with: ./build.sh manifests/1.3.0/opensearch-1.3.0.yml --component alerting
```

### To reproduce
Run `./build.sh manifests/1.3.0/opensearch-1.3.0.yml --component alerting --snapshot` in the root of this build repo inside the CI container.

**Comment by qreshi** [comment]
I was able to reproduce this with `./build.sh manifests/1.3.0/opensearch-1.3.0.yml` from `opensearch-build` but I also tracked down the problematic call to the [./gradlew assemble](https://github.com/opensearch-project/alerting/blob/main/scripts/build.sh#L64) call in Alerting's `scripts/build.sh`

The call with the arguments all filled out is the following:
```
./gradlew assemble --no-daemon --refresh-dependencies -DskipTests=true -Dopensearch.version=1.3.0 -Dbuild.snapshot=false -x ktlint
```

Strangely enough, when you remove the `-Dopensearch.version` argument, the assemble passes.

Given that the default on main for the version  (in `build.gradle`) is:
```
opensearch_version = System.getProperty("opensearch.version", "1.3.0-SNAPSHOT")
```

And we then do:
```
version = "${opensearch_version}" - "-SNAPSHOT" + ".0"
if (isSnapshot) {
    version += "-SNAPSHOT"
}
```

It should still be ending up with the same version internally (`1.3.0`). Knowing this, is anyone able to reproduce this with another plugin?

**Comment by tianleh** [comment]
@qreshi I just tried removing alerting from `manifests/1.3.0/opensearch-1.3.0.yml` and re-run `./ci.sh manifests/1.3.0/opensearch-1.3.0.yml --snapshot` which now succeeds.

If the issue of alerting cannot be fixed soon, could we for now remove alerting from `manifests/1.3.0/opensearch-1.3.0.yml` (and the test manifest)? This can unblock others who are updating other manifest files. The PR checks are failing due to the alerting error.

**Comment by qreshi** [comment]
@tianleh Yeah you can remove it for now if it's blocking anything. Although, doesn't Alerting publish the Notification subproject locally before moving on to Index Management? I tried removing Alerting from the manifest locally and when I re-ran `./build.sh manifests/1.3.0/opensearch-1.3.0.yml`, it failed on Index Management.

**Comment by qreshi** [comment]
Also, I'm not really a gradle expert but looking up the gradle code, I saw [this](https://github.com/gradle/gradle/blob/5ec3f672ed600a86280be490395d70b7bc634862/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/DefaultRepositoryContentDescriptor.java#L62-L68) was where that error message was being generated from.

We actually have a reference to the repos in a couple places. I tried removing all references to [repositories.gradle](https://github.com/opensearch-project/alerting/blob/main/build-tools/repositories.gradle) in the Alerting project and then the error became:

```
* What went wrong:
A problem occurred evaluating root project 'opensearch-alerting'.
> A problem occurred configuring project ':alerting'.
   > Cannot mutate content repository descriptor 'MavenRepo' after repository has been used
```

Perhaps someone with Gradle expertise can identify if this is a configuration problem.

**Comment by tianleh** [comment]
> @tianleh Yeah you can remove it for now if it's blocking anything. Although, doesn't Alerting publish the Notification subproject locally before moving on to Index Management? I tried removing Alerting from the manifest locally and when I re-ran `./build.sh manifests/1.3.0/opensearch-1.3.0.yml`, it failed on Index Management.

I will wait for this issue to be resolved given the complicated situation.

**Comment by tianleh** [comment]
Based on my observation, the CI command`./ci.sh manifests/1.3.0/opensearch-1.3.0.yml --snapshot` was successful on 2/19/2022 12pm but failed on 2/21/2022 12pm. Seems something happened between 2/19/2022 12pm and 2/21/2022 12pm. Hope this can help narrow down the scope of research a bit.
